Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2
# Necro Government and Undead Flavor

## Changelog:
- Added Necro Government:
  - No vassal contracts, taxes and levies based on opinion.
  - Only rulers of the same religious family can inherit.
  - Dark Frenzy instead of Crown Authority.
    - By default, you're 50% of normal government but can be up to 150% by keeping high Frenzy.
    - High Frenzy gives you troop, tax, and development bonuses.
    - Frenzy decays over time so to be efficient they should do their mechanics.
  - Get Frenzy on converting provinces to their culture and religion.
  - Get Frenzy on winning armies, sieging, raiding.
  - Killed knights become yours as undead. High Frenzy increases the chance.
    - Becoming an undead, the character cuts all his previous relationships, abdicate, and becomes disinherited.
    - The character's previous relatives, friends, lovers get stress as the character died if their faith shuns or considers the undead criminals.
    - Reinforce armies much faster.
- Disinheriting undead is free if your faith shuns or considers them criminals.
- The Faiths of the Death religious family use black, purple, dark blue, white, grey, light red colors for CoAs.
- The Grim Religions can't do hunts and feasts.
- Northrend frozen landscape appears on event backgrounds in Icecrown, Storm Peaks, and Dragonblight.
- Rulers of the Scourge are called Lich Kings/Queens.


## Developer changelog:
- Added many triggers and effects to handle monarchic governments that should work kinda the same.
- Uncommented `debug_start_great_holy_war_interaction`.

# Tasks:

<details>
<summary>Click to expand</summary>

- [x] Create game concepts for new governments and Dark Governments
- [x] AI should disinherit undead, demons, etc. for free and often
- [x] Relocalize this:
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/122537363-88fc9c80-d036-11eb-95ee-713ef0baa81b.png)

</details>

- [x] Government resets on losing requirements
- [x] Adapt `StatecraftGetVassalType`.
- [x] Fix not having Crown Authority
- [x] Add progress to Dark Frenzy
- [x] Add Dark Frenzy icons
- [x] Add Dark Frenzy modifiers
- [x] Create game concept for Dark Frenzy
- [x] Send interface message on getting new Dark Frenzy law
- [x] Switching from Holy Light to Helya sets Necro government
- [x] Switching to Light clears Dark Frenzy
- [x] Change this background
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/122667213-b6765100-d1c2-11eb-9e60-be9c5034c7f6.png)

</details>

- [x] Dark Frenzy progress bar should call tooltip
- [x] Localize Dark Frenzy scripted effects
- [x] Explain why you can't change laws of Dark Frenzy yourself
- [x] Check authority change effect
- [x] Angerboda starting Undead
- [x] Grim Religions shouldn't have feasts, hunts
- [x] Handle death in `combat_event.1001`
- [x] Handle notification in `combat_event.1001`
- [x] Add loyalty opinion
- [x] Give stress/event if your friend/relative turned undead, #511
- [x] Check errors in battle, knight, combat, death events
- [x] Increase chance with Frenzy
- [x] Undead resets all character's relationships
- [x] Check if relatives get notification for game start undead
- [x] Check soulmate, secret lover, best friend
- [x] Event backgrounds
- [x] Replace `being_is_illegal_or_shunned_trigger` with `trait_is_shunned_or_criminal_in_faith_trigger`.
- [x] It causes longer loading?
- [x] Crusader realms (created in crusade)
- [x] Scourge shouldn't be claimed
<details>
<summary>Click to expand</summary>

![image](https://user-images.githubusercontent.com/46384992/125998125-9d729800-4400-4e74-abdf-1aebc42a417a.png)

</details>

- [ ] Mention that Dark Religions can't hunt and feast
- [x] Convert to Holy Light, kill, inherit by the undead, check if you get yearly Dark Frenzy

</details>


# How to test:
Start as a kvaldir in any date or as a vassal of the Scourge in the last 2 dates. Test the mentioned in the changelog features.

@Warcraft-GoA-Development-Team/english-proofreaders open the [Page](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/pull/463/files?file-filters%5B%5D=.yml), scroll to the bottom to the `.yml` files, check them, suggest changes. If you forgot, here are the [Suggestion Guidelines](https://github.com/Warcraft-GoA-Development-Team/Warcraft-Guardians-of-Azeroth-2/wiki/How-to-be-Tester-(For-CK3)#reviewing-code).
